packages/@sanity/form-builder/src/inputs/files/common/FileInputButton/FileInputButton.tsx

```diff
@@ -0,0 +1,92 @@
+// todo: consider moving this into @sanity/ui and make it exposed as a primitive
+
+import React from 'react'
+import {useId} from '@reach/auto-id'
+import styled, {css} from 'styled-components'
+import {Button, Theme} from '@sanity/ui'
+
+import {focusRingBorderStyle, focusRingStyle} from '../focusringUtils'
+
+type Props = Omit<React.ComponentProps<typeof Button>, 'type' | 'value' | 'onSelect'> & {
+  accept?: string
+  capture?: 'user' | 'environment'
+  multiple?: boolean
+  onSelect?: (files: File[]) => void
+  children?: React.ReactNode
+}
+
+const Input = styled.input`
+  overflow: hidden;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  opacity: 0;
+  position: absolute;
+`
+
+// Note: this is rendered as a label
+const SelectButton = styled(Button)(({theme}: {theme: Theme}) => {
+  const border = {width: 1, color: 'var(--card-border-color)'}
+
+  return css`
+    //---
+    // Needed because normalize.css in the studio overrides this in Safari.
+    // Safe to remove the following line when when we no longer use normalize.css
+    appearance: none !important;
+    //----
+
+    &:not([data-disabled='true']) {
+      &:focus-within {
+        box-shadow: ${focusRingStyle({
+          base: theme.sanity.color.base,
+          border,
+          focusRing: theme.sanity.focusRing,
+        })};
+      }
+      &:focus:not(:focus-visible) {
+        box-shadow: ${focusRingBorderStyle(border)};
+      }
+    }
+
+    // This is a little hack to fix wonky @sanity/ui Button rendering where there's a child element.
+    // This issue should be fixed in @sanity/ui and this hack removed from here.
```

Comment:
Did we create an issue for this under the Sanity UI repo or in the internal issue tracker?

Reply:
Good point! Created https://github.com/sanity-io/design/issues/471 and updated the comment to point at it.